apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -227,6 +227,11 @@ type OpenTelemetryCollectorSpec struct {
 	// This is only relevant to statefulset, and deployment mode
 	// +optional
 	TopologySpreadConstraints []v1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+
+	// ConfigMaps is a list of ConfigMaps in the same namespace as the OpenTelemetryCollector
+	// object, which shall be mounted into the Collector Pods.
+	// Each ConfigMap will be added to the Collector's Deployments as a volume named `configmap-<configmap-name>`.
+	ConfigMaps []ConfigMapsSpec `json:"configmaps,omitempty"`
```

Comment:
If Prometheus sets a precedence for not letting this be configurable then I am ok following that.  The most robust solution though would be to allow it to be configurable and fallback to a default value if no path is provided.

Reply:
I don't like the prometheus-operator solution. I'd argue it makes things more complex for both the user and the operator code. The user, because they need to both read the documentation and then ensure they're setting the value correctly wherever they need the file name. And for the operator, because it now has to take care of generating the mount path, and the how the path is generated is now part of our public API.

As opposed to the user just setting whatever they want and the operator applying it without any additional checks. Arguably this can lead to footguns where the mount overrides something important, but I think it's the user's responsibility to make sure this doesn't happen - and this may even be a feature in some circumstances.

All that said, if we're following prometheus-operator conventions everywhere else, we should probably do it here as well for consistency if nothing else.

Reply:
I like this argument :) I think making it configurable with a default if not set seems like a good middle ground here. Does that work for you @swiatekm-sumo?

Reply:
Making it configurable with a default comes with the worst of both worlds for us, but the best for the users, so I'm fine with it.

Reply:
What was the outcome of this discussion?  I don't see us using a default value for the path if it is not configured.

Reply:
I believe the suggestion to make the configMaps' mountPaths configurable is the solution we found better for the users. So, I have changed the PR accordingly.

Reply:
If the user doesn't configure the mountPath or name what happens? Do we need a validating case for that?

Reply:
I don't think so, I would say both are required to mount extra configMaps.

Reply:
If they are required we should validate that requirement.

Reply:
@TylerHelmuth, I have changed the requirements and added some validations to the configmaps's definition.
When you get a chance could you please have a look?